openhcl/hcl/src/ioctl.rs

```diff
@@ -1521,57 +1692,116 @@ impl<T> Drop for ProcessorRunner<'_, T> {
 }
 
 impl<'a, T: Backing> ProcessorRunner<'a, T> {
-    fn set_reg(&mut self, regs: &[HvRegisterAssoc]) -> Result<(), Error> {
+    // These registers are handled specially by the kernel through a dedicated
+    // ioctl. is_kernel_managed is arch-specific to guard against an into() on
+    // an HvArmRegisterName that overlaps one of these x86-specific values.
+    #[cfg(guest_arch = "x86_64")]
+    fn is_kernel_managed(&self, name: HvX64RegisterName) -> bool {
+        if name == HvX64RegisterName::Xfem {
+            if self.hcl.isolation() == IsolationType::Tdx {
+                return true;
+            }
+        }
+
+        is_vtl_shared_mtrr(name)
+            || matches!(
+                name,
+                HvX64RegisterName::Dr0
+                    | HvX64RegisterName::Dr1
+                    | HvX64RegisterName::Dr2
+                    | HvX64RegisterName::Dr3
+                    | HvX64RegisterName::Dr6
```

Comment:
Don't we need to special case dr6?

Reply:
Ah, indeed. (I was going to say that the kernel currently implements that special handling in its dedicated get/set-register ioctl, but that ioctl isn't VTL-aware, so we'll want to have VTL-aware dr6 handling in OpenHCL.)